ecr/ref.go

```diff
@@ -104,6 +150,26 @@ func parseARN(a string) (ECRSpec, error) {
 
 }
 
+// parseARNResource parses the resource segment of an ECR ARN and returns the tag/digest (object) and removes the
+// tag/digest (object) from the resource segment.
+// An example of an ECR ARN resource is "repository/myimage:mytag" and the object returned is "mytag"
+// Another example is "repository/myimage@sha256:47bfdb88c..." and the object returned is "@sha256:47bfdb88c..."
+func parseARNResource(resource *string) string {
```

Comment:
This isn't really related to ARNs, so I'd probably call it `parseObject` as that's what it returns for you.